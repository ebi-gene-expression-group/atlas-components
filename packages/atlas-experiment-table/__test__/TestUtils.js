import React from 'react'
import styled from 'styled-components'

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const TableCellDiv = styled.div`
  font-size: 13px;
  font-family: Helvetica, Arial, FreeSans, "Liberation Sans", sans-serif;
`

const tableHeader = [
  { type: `sort`, title: `Loaded date`, width: 140, dataParam: `lastUpdate` },
  { type: `search`, title: `species`, width: 200, dataParam: `species` },
  { type: ``, title: `experiment description`, width: 360, dataParam: `experimentDescription`, link: `experimentAccession`, resource: `experiments`, endpoint: `Results` },
  { type: `plain`, title: `experiment factors`, width: 260, dataParam: `experimentalFactors` },
  { type: `sort`, title: `Number of assays`, width: 160, dataParam: `numberOfAssays`, link: `experimentAccession`, resource: `experiments`, endpoint: `Experiment Design` },
]

const data = [
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-EHCA-2`,
    experimentDescription: `Melanoma infiltration`,
    lastUpdate: `16-11-2018`,
    numberOfAssays: 6638,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`, `sampling site`,`time`],
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-GEOD-99058`,
    experimentDescription: `Single cell`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 250,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`]
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-MTAB-5061`,
    experimentDescription: `healthy individuals and type 2 diabetes patients`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 3514,
    numberOfContrasts: 0,
    species: `Homo sapiens`,
    kingdom: `plants`,
    experimentalFactors: [`single cell identifier`, `disease`],
  }
]

export {getRandomInt, TableCellDiv, tableHeader, data}
