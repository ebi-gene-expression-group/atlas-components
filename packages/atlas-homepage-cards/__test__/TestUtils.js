

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const dummyCards = () => {
  return [
    {
      iconType: `species`,
      iconSrc: `mus musculus`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `species`,
      iconSrc: `homo sapiens`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `unknown`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `experiments`,
      iconSrc: `http://www.zhao.com`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `image`,
      iconSrc: `logo-goes-here`,
      iconDescription: `An image card`,
      content: [
        {
          text: `Go to experiments`,
        }
      ]
    },
    {
      iconType: `imagespecies`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
  ]
}

export {getRandomInt, dummyCards}