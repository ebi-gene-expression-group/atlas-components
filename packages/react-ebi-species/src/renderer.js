"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
