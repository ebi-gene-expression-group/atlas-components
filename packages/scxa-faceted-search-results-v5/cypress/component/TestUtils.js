import facets from '../fixtures/vindicatorsResponse.json'
import organismParts from '../fixtures/organismPartsResponse.json'
import cellTypes from '../fixtures/cellTypesResponse.json'
import species from '../fixtures/speciesResponse.json'

import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";
import CheckboxFacetGroupsDefaultProps from "../../src/facetgroups/CheckboxFacetGroupsDefaultProps";
import MultiSelectDropdownFacetGroupsDefaultProps
  from "../../src/facetgroups/MultiSelectDropdownFacetGroupsDefaultProps";

const getPropsForCheckBoxGroupWithTooltip = () => {
  return Cypress._.shuffle(CheckboxFacetGroupsDefaultProps).pop()
}

const getPropsForMultiSelectDropdownGroupWithTooltip = () => {
  return Cypress._.shuffle(MultiSelectDropdownFacetGroupsDefaultProps).pop()
}

const getPropsForMarkerGeneFacet = () => {
  const markerGeneName = `Marker genes`;
  return CheckboxFacetGroupsDefaultProps.find(facet => facet.name === markerGeneName)
}

const getPropsWithoutTooltip = () => {
  let props = getPropsForCheckBoxGroupWithTooltip()
  props.description = ``
  return props
}

const getFacets = () => {
  return Cypress._.shuffle(facets).filter(() => Math.random() > 0.5)
}

const getRandomSpecies = () => {
  return Cypress._.shuffle(species).filter(() => Math.random() > 0.5)
}

const getRandomOrganismParts = () => {
  return Cypress._.shuffle(organismParts).filter(() => Math.random() > 0.5)
}

const getRandomCellTypes = () => {
  return Cypress._.shuffle(cellTypes).filter(() => Math.random() > 0.5)
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { getRandomInt, getFacets, getRandomOrganismParts, getRandomCellTypes, getRandomSpecies,
  getPropsForCheckBoxGroupWithTooltip, getPropsForMultiSelectDropdownGroupWithTooltip,
  getPropsWithoutTooltip, getPropsForMarkerGeneFacet,
  ExperimentTableCard, ExperimentTableHeader }
