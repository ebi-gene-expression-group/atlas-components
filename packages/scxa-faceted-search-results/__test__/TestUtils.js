import React from 'react'
import PropTypes from 'prop-types'

const EpisodeCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

EpisodeCard.propTypes = {
  title: PropTypes.string.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min);
  max = Math.floor(max);
  return Math.floor(Math.random() * (max - min)) + min; //The maximum is exclusive and the minimum is inclusive
}

const vindicators = [
  { group:`Vindicator`, value: `supernova`, label: `Supernova`},
  { group:`Vindicator`, value: `noob-noob`, label: `Noob-Noob` },
  { group:`Vindicator`, value: `vance_maximus`, label: `Vance Maximus`},
  { group:`Vindicator`, value: `alan_rails`, label: `Alan Rails`},
  { group:`Vindicator`, value: `crocubot`, label: `Crocubot` },
  { group:`Vindicator`, value: `million_ants`, label: `Million Ants`},
  { group:`Vindicator`, value: `morty_smith`, label: `Morty Smith`},
  { group:`Vindicator`, value: `rick_sanchez`, label: `Rick Sanchez`},
  { group:`Vindicator`, value: `lady_katana`, label: `Lady Katana`},
  { group:`Vindicator`, value: `calypso`, label: `Calypso`},
  { group:`Vindicator`, value: `diablo_verde`, label: `Diablo Verde`}
]

const episodes = [
  {
    element: {
      title: "Raising Gazorpazorp"
    },
    facets: [
      {
        group: "Planet",
        value: "gazorpazorp",
        label: "Gazorpazorp"
      },
      {
        group: "Guest character",
        value: "gwendolyn",
        label: "Gwendolyn"
      },
      {
        group: "Guest character",
        value: "ma-sha",
        label: "Ma-Sha"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "The wedding squanchers"
    },
    facets: [
      {
        group: "Planet",
        value: "squanch",
        label: "Squanch"
      },
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Season",
        value: "2",
        label: "2"
      }
    ]
  },
  {
    element: {
      title: "The Rickshank redemption"
    },
    facets: [
      {
        group: "Guest character",
        value: "birdperson",
        label: "Birdperson"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "3",
        label: "3"
      }
    ]
  },
  {
    element: {
      title: "Ricksy business"
    },
    facets: [
      {
        group: "Guest character",
        value: "squanchy",
        label: "Squanchy"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  },
  {
    element: {
      title: "Close Rick-counters of the Rick kind"
    },
    facets: [
      {
        group: "Guest character",
        value: "ricktiminus_sancheziminius",
        label: "Ricktiminus Sancheziminius"
      },
      {
        group: "Guest character",
        value: "abradolf_lincler",
        label: "Abradolf Lincler"
      },
      {
        group: "Planet",
        value: "buttworld",
        label: "Buttworld"
      },
      {
        group: "Season",
        value: "1",
        label: "1"
      }
    ]
  }
]



export {getRandomInt, vindicators, episodes, EpisodeCard}
