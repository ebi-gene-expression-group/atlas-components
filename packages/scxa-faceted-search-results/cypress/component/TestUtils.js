import episodes from '../fixtures/episodes.json'
import PropTypes from "prop-types";
import styled from "styled-components";
import React from "react";

const facets = () => {
  return episodes.reduce((acc, episode) => acc.concat(episode.facets), [])
}

const getUniqueFacets = () => {
  const allFacets = facets()
  return allFacets
    .filter((facet, index) => allFacets.findIndex((thatFacet) => facet.value === thatFacet.value) === index)
    .map((facet) => ({
      ...facet,
      disabled: false
    }))
}

const getFacetTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => facet.description)
}

const getFacetWithoutTooltip = () => {
  return Cypress._.shuffle(getUniqueFacets()).find(facet => !facet.description)
}

const getPropsWithTooltip = (facetTooltip) => {
  return {
    facetGroupName: facetTooltip.group,
      facetGroupNameDescription: facetTooltip.description,
    facets: getUniqueFacets().filter(facet => facet.group === facetTooltip.group),
    onChange: () => {}
  }
}

const getPropsWithoutTooltip = () => {
  const facetWithoutTooltip = getFacetWithoutTooltip()
  return {
    facetGroupName: facetWithoutTooltip.group,
      facets: getUniqueFacets().filter(facet => facet.group === facetWithoutTooltip.group),
    onChange: () => {}
  }
}

const ExperimentTableCard = ({title}) =>
  <div>
    <p>{title}</p>
  </div>

ExperimentTableCard.propTypes = {
  title: PropTypes.string.isRequired
}

const IconDiv = styled.div`
  width: 15%;
  text-align: center;
`
IconDiv.displayName = 'IconDiv'

const BitDiv = styled.div`
  width: 25%;
  text-align: center;
`
const DoveDiv = styled.div`
  width: 25%;
  text-align: center;
`

const ExperimentTableHeader = ({onClick}) =>
  <DoveDiv>
    <IconDiv onClick={()=>onClick(`test`)}>
      Meow
    </IconDiv>
    <BitDiv onClick={()=>onClick(`test`)}>
      Wow wow
    </BitDiv>
  </DoveDiv>

ExperimentTableHeader.propTypes = {
  onClick: PropTypes.func.isRequired
}

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

export { episodes, getRandomInt, getUniqueFacets, getFacetTooltip, getPropsWithTooltip, getPropsWithoutTooltip,
  ExperimentTableCard, ExperimentTableHeader }
