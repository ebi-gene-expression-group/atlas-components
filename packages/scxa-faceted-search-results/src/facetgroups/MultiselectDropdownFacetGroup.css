/* Unfortunately we need to set some EBI VF overrides here because it applies styles to elements deper than
React-Selectâ€™s APIs (e.g. the div that encloses the input rather than the input itself) */
.input-clear input, .input-clear input:focus {
  height: unset;
  box-shadow: none;
  margin: 0;
}
