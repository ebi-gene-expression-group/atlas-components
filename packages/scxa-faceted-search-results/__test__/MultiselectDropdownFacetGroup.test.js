import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import _ from 'lodash'

import { getRandomInt, episodes } from './TestUtils'

import MultiselectDropdownFacetGroup from '../src/facetgroups/MultiselectDropdownFacetGroup'

Enzyme.configure({ adapter: new Adapter() })

describe(`MultiselectDropdownFacetGroup`, () => {
  const allFacets = episodes.reduce((acc, episode) => acc.concat(episode.facets), [])
  const uniqueFacets =
    allFacets
      .filter((facet, index) => allFacets.findIndex((thatFacet) => facet.value === thatFacet.value) === index)
      .map((facet) => ({
        ...facet,
        disabled: false
      }))

  const facetTooltip = _.shuffle(uniqueFacets).find(facet => facet.description)
  const facetWithoutTooltip = _.shuffle(uniqueFacets).find(facet => !facet.description)

  const propsWithTooltip = {
    facetGroupName: facetTooltip.group,
    facetGroupNameDescription: facetTooltip.description,
    facets: uniqueFacets.filter(facet => facet.group === facetTooltip.group),
    onChange: () => {}
  }

  const propsWithoutTooltip = {
    facetGroupName: facetWithoutTooltip.group,
    facets: uniqueFacets.filter(facet => facet.group === facetWithoutTooltip.group),
    onChange: () => {}
  }

  test(`displays tooltip if it exists`, () => {
    const groups = [...new Set(uniqueFacets.map((facet) => facet.group))]
    const descriptions = [...new Set(uniqueFacets.map((facet) => facet.description))]
    const randomMultiselectFacetGroup = groups[getRandomInt(0, groups.length)]
    const randomMultiselectFacetGroupDescription = descriptions[getRandomInt(0, descriptions.length)]

    const wrapper = mount(<MultiselectDropdownFacetGroup {...propsWithTooltip} />)
    expect(wrapper.find(`sup`).first().html()).toEqual(expect.stringMatching(`data-tip`))
  })

  test(`doesnâ€™t display tooltip if not present`, () => {
    const wrapper = mount(<MultiselectDropdownFacetGroup {...propsWithoutTooltip} />)
    expect(wrapper.find(`sup`).exists()).toEqual(false)
  })

  test(`matches snapshot`, () => {
    const tree =
      renderer
        .create(
          <MultiselectDropdownFacetGroup
            facetGroupName = {uniqueFacets[0].group}
            facetGroupNameDescription = {uniqueFacets[0].description}
            facets = {uniqueFacets.filter(facet => facet.group === uniqueFacets[0].group)}
            onChange = {() => {}} />)
        .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
