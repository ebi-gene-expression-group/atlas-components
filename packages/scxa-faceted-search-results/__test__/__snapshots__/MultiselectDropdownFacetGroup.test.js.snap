// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiselectDropdownFacetGroup matches snapshot 1`] = `
<div
  className="padding-bottom-xlarge"
>
  <h4>
    Planet
    <sup>
      <span
        className="icon icon-generic"
        data-html={true}
        data-icon="i"
        data-tip="<span>Planets where events in this episode take<br>place</span>"
        style={
          Object {
            "color": "lightgrey",
            "fontSize": "smaller",
          }
        }
      />
    </sup>
  </h4>
  <div
    className="css-1pcexqc-container"
    onKeyDown={[Function]}
  >
    <div
      className="css-12emboz"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3"
      >
        <div
          className="css-151xaom-placeholder"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="facetGroupMultiSelectDropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6"
      >
        <span
          className="sc-bxivhb iCJrRm"
          cx={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
